src/CityOnPlanet.cpp

```diff
@@ -121,19 +126,89 @@ void CityOnPlanet::LookupBuildingListModels(citybuildinglist_t *list)
 	assert(!models.empty());
 	Output("Got %d buildings of tag %s\n", static_cast<int>(models.size()), list->modelTagName);
 	list->buildings = new citybuilding_t[models.size()];
-	list->numBuildings = models.size();
+	list->numBuildings = (uint32_t)models.size();
+	numberofSimplebuildings = 0;
+
+	Json data = JsonUtils::LoadJsonDataFile(CITY_CONFIG_FILE);
+
+	// allow config to determine city sizes
+	config_atmo_size = DEFAULT_ATMO_SIZE;
+	config_airless_size = DEFAULT_AIRLESS_SIZE;
+	if (!data["global"].is_null()) {
+		config_atmo_size = data["global"].value("atmo-size", DEFAULT_ATMO_SIZE);
+		config_airless_size = data["global"].value("airless-size", DEFAULT_AIRLESS_SIZE);
+
+		population2radius[0] = data["global"].value("pop-0", 1000000000.0);
+		population2radius[1] = data["global"].value("pop-1", 200000000.0);
+		population2radius[2] = data["global"].value("pop-2", 20000000.0);
+		population2radius[3] = data["global"].value("pop-3", 1000000.0);
+		population2radius[4] = 0.0;
+
+		cityradius_mod[0] = data["global"].value("mod-0", 1.0);
+		cityradius_mod[1] = data["global"].value("mod-1", 0.8);
+		cityradius_mod[2] = data["global"].value("mod-2", 0.6);
+		cityradius_mod[3] = data["global"].value("mod-3", 0.4);
+		cityradius_mod[4] = data["global"].value("mod-4", 0.2);
+
+		cityradius_rnd[0] = data["global"].value("rnd-0", 30);
+		cityradius_rnd[1] = data["global"].value("rnd-1", 15);
+		cityradius_rnd[2] = data["global"].value("rnd-2", 14);
+		cityradius_rnd[3] = data["global"].value("rnd-3", 12);
+		cityradius_rnd[4] = data["global"].value("rnd-4", 10);
```

Comment:
Similarly to the prior comment, please prefer loading this from JSON as an array of objects into an array of structs, rather than a manually-suffixed set of fields. If the defaults are significant, please move them to a code comment on the appropriate struct definition and/or fill any missing fields in the data file.